### ISSUE TITLE
N805 False positive for `class Meta(type(base))`

### ISSUE THREAD

**Issue by randolf-scholz** [astral-sh/ruff#12736]
```python
from typing import Protocol

class MyMeta(type):
    def __subclasscheck__(cls, other): ...  # ✅

class MyProtocolMeta(type(Protocol)):
    def __subclasscheck__(cls, other): ...  # ❌
```

https://play.ruff.rs/5573b0e2-fba5-47c1-bc69-a7e780324a86

**Comment by randolf-scholz** [comment]
Problem is with this function, which does not correctly recognizes metaclass in this case:

https://github.com/astral-sh/ruff/blob/df7345e118f456e43b04aefbbaaa253c16b62329/crates/ruff_python_semantic/src/analyze/class.rs#L115-L122
